stetho-sample/src/debug/java/com/facebook/stetho/sample/SampleDebugApplication.java

```diff
@@ -42,7 +53,60 @@ private void initializeStetho(final Context context) {
                 .finish();
           }
         })
-        .enableWebKitInspector(Stetho.defaultInspectorModulesProvider(context))
+        .enableWebKitInspector(new ExtInspectorModulesProvider(context))
         .build());
   }
+
+  private static class ExtInspectorModulesProvider implements InspectorModulesProvider {
+
+    private Context mContext;
+
+    ExtInspectorModulesProvider(Context context) {
+      mContext = context;
+    }
+
+    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+    @Override
+    public Iterable<ChromeDevtoolsDomain> get() {
+      Stetho.DefaultInspectorModulesBuilder builder = new Stetho.DefaultInspectorModulesBuilder(mContext);
+      if (Build.VERSION.SDK_INT >= DatabaseConstants.MIN_API_LEVEL) {
+
+        // sample calendars content provider we want to support
+        ContentProviderSchema calendarsSchema = new ContentProviderSchema.Builder()
+            .table(new Table.Builder()
+                .uri(CalendarContract.Calendars.CONTENT_URI)
+                .projection(new String[] {
+                    CalendarContract.Calendars._ID,
+                    CalendarContract.Calendars.NAME,
+                    CalendarContract.Calendars.ACCOUNT_NAME,
+                    CalendarContract.Calendars.IS_PRIMARY,
+                })
+                .build())
+            .build();
+
+        // sample events content provider we want to support
+        ContentProviderSchema eventsSchema = new ContentProviderSchema.Builder()
+            .table(new Table.Builder()
+                .uri(CalendarContract.Events.CONTENT_URI)
+                .projection(new String[]{
+                    CalendarContract.Events._ID,
+                    CalendarContract.Events.TITLE,
+                    CalendarContract.Events.DESCRIPTION,
+                    CalendarContract.Events.ACCOUNT_NAME,
+                    CalendarContract.Events.DTSTART,
+                    CalendarContract.Events.DTEND,
+                    CalendarContract.Events.CALENDAR_ID,
+                })
+                .build())
+            .build();
+
+        Database database = new Database();
```

Comment:
Sorry for the late review, I was working on getting stetho-1.2.0 out with a much improved API to support this kind of customization.  I think we can now approach this with a new, much stronger API:

    new DefaultInspectorModulesProvider()
      .provideDatabaseDriver(new ContentProviderDatabaseDriver(...))
      .finish();

This implies that the default provider is install installed. If the default is to be configured, they can use the existing (as of 1.2.0) method .databaseFiles(...) and just provide a no-op files lister.